src/erl_cache_server.erl

```diff
@@ -279,16 +279,20 @@ check_mem_usage( Name, CurrentWords ) ->
     ok.
 
 %% @private
--spec do_apply(mfa() | function()) -> term().
+-spec do_apply(function() | mfa() | {function(), [any()]}) -> term().
 do_apply({M, F, A}) when is_atom(M), is_atom(F), is_list(A) ->
     apply(M, F, A);
+do_apply({F, A}) when is_function(F, length(A)) ->
+    apply(F, A);
 do_apply(F) when is_function(F) ->
     F().
 
 %% @private
 -spec is_error_value(erl_cache:is_error_callback(), erl_cache:value()) -> boolean().
 is_error_value({M, F, A}, Value) ->
     apply(M, F, [Value|A]);
+is_error_value({F, A}, Value) ->
```

Comment:
should this be guarded too?

Reply:
There were no guards in this function previously like in `do_apply` so I didn't add them (tbh, I'm not sure if they're even needed in `do_apply` since data is verified during input).